src/Compilers/CSharp/Portable/Compiler/EqualityMethodBodySynthesizer.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Emit
+{
+    internal static class EqualityMethodBodySynthesizer
+    {
+        public static BoundExpression GenerateEqualsComparisons<TList>(
+            NamedTypeSymbol containingType,
+            BoundExpression receiverAccess,
```

Comment:
>receiverAccess [](start = 28, length = 14)

Might be more clear to call this `otherAccess` (i.e. the other thing to which we are comparing this).